### PR TITLE
Token Holders Scanner upgrade (now updates the token holders balances)

### PR DESCRIPTION
**Pull request by lucasmenendez**
- The issue on new tokens creation when the database has a token already registered was fixed.
- The issue when two tokens have common token holders was fixed.
- The issue with freeze tokens when were no new logs on new blocks was fixed.
- The issue with contexts and large bulks of token holders to save in the database was fixed.
- First version of balances tracking for token holders.

### CODE REVIEW

**Comment by lucasmenendez** [comment]
@jordipainan please, start to review from [7714e1c](https://github.com/vocdoni/census3/pull/13/commits/7714e1cba0d2d21eb624aad2c1c99166354c250d). The previous commmits are already reviewed in the last PR

File: api/holders.go [hunk 1]
```diff
@@ -1,80 +1,93 @@
...
+func (capi *census3API) countHolders(msg *api.APIdata, ctx *httprouter.HTTPContext) error {
+	ctx2, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	addr := common.HexToAddress(ctx.URLParam("address"))
+	numberOfHolders, err := capi.sqlc.CountTokenHoldersByTokenID(ctx2, addr.Bytes())
+	if err != nil {
+		if errors.Is(sql.ErrNoRows, err) {
+			log.Errorf("no holders found for address %s: %w", addr, err)
+			return ErrNoFoundTokenHolders.Withf("token address: %s", addr)
```
Comment (jordipainan) [inline api/holders.go:77]: Do you want to treat this as an error? I think it is a bit misleading, the call in this case resolved fine but no holders were found. We can return a 200 with an empty list and a message saying "no holders found"

**Review by jordipainan — APPROVED** [review]

File: api/holders.go (hunk 1 above)
Comment (lucasmenendez) [reply api/holders.go:77]: Yes, I am agree. Anyway I am not sure if the current token holders endpoints are necessary or not (probably not), but I will update them before merge.

Comment (lucasmenendez) [reply api/holders.go]: I solve it returning `204 No content`, what do you think @jordipainan?

**Pull request merged**